spyder/plugins/projects/widgets/projectdialog.py

```diff
@@ -147,6 +149,8 @@ def _validate_location(
             reasons["location_not_writable"] = True
         elif name is not None:
             project_path = osp.join(location, name)
+            if os.name == "nt" and re.search(r":", name):
+                reasons["wrong_name"] = True
```

Comment:
```suggestion
                # Prevent creating a project in directory with colons.
                # Fixes spyder-ide/spyder#16942
                reasons["wrong_name"] = True
```

Add clarifying comment.